compiler/src/jlang/util/JLangDesugared.java

```diff
@@ -50,12 +50,13 @@ public JLangDesugared(Job job) {
                 // should not create qualified Special nodes.
                 new DesugarInnerClasses(job, ts, nf),
 
+                // Translate synchronized keyword to a try-finally block to handle control transfer.
```

Comment:
Previously, I used `This` instead of `UnqualifiedThis` when desugaring `synchronized` so I moved `DesugarSynchronized` before `DesugarInstanceInitializers`. Later, I realized that `UnqualifiedThis` was the correct way to go and forgot to move it back. I moved it after `DesugarInstanceInitializers` in the latest commit.